docs/contributing/versioning.md

```diff
@@ -0,0 +1,121 @@
+# Versioning
+
+With an user interface (UI) library there are many ways that people can consume our code.
+
+We follow [semantic versioning](https://semver.org/) but a UI library often has subjective changes such as visual spacing changes.
+
+This document aims to give a basis to make decisions as a team.
+
+## Contents
+
+- [Stability](#stability)
+- [Proposal](#proposal)
+- [Deprecation](#deprecation)
+- [Migration](#migration)
+- [Public API](#public-api)
+- [Accidental breaking changes](#accidental-breaking-changes)
+
+## Stability
+
+A stable library prioritises the users and ecosystem that it supports.
+
+We often release new components and features which will encourage people to update.
+
+But we should ensure that we only make breaking changes when it is absolutely necessary.
+
+Good reasons to make breaking changes:
+
+- issues that are barriers end-users (users of services) based on evidence (for example user research)
+- issues that are barriers users (users of GOV.UK Frontend) based on evidence
+- accessibility issues
+- security issues
+- performance issues
+
+Bad reasons to make breaking changes:
+
+- you think you can sneak a change in with other breaking changes - breaking changes need to be migrated from, which is effort for our users we should try to avoid
+- changing the name of an API, based on a hunch
+- adopting a technology based on popularity and not because of the problems they solve
+- changing a components interface without a good reason to do so
+
+This is similar to how we try to tackle most problems, by focusing on user needs first.
+
+Ember.JS is a great example of prioritising stability see [Stability without Stagnation: Lessons Learned from Shipping Ember.js
+](https://speakerdeck.com/wycats/stability-without-stagnation-lessons-learned-from-shipping-ember-dot-js)
+
+We do not want our users to be reliant on certain versions of our library with no way forward.
+
+Whenever we decide to make a breaking change we must ensure we do our best to try to first deprecate the feature and allow a migration path.
+
+## Proposal
+
+Some breaking changes that need discussion may first be proposed in the [GOV.UK Design System architecture repository](https://github.com/alphagov/govuk-design-system-architecture/blob/master/proposals/README.md).
+
+This is to ensure the community can get involved with the decision.
+
+You should make sure to take steps to actively involved the community and not wait for feedback, this might be in the form of presentations or meetings. Since these issues may be difficult to understood without a conversation.
+
```

Comment:
Could this paragraph be clarified a little bit, I'm not sure what it's trying to say.